crates/runtime/src/table.rs

```diff
@@ -212,6 +212,33 @@ impl Table {
         }
     }
 
+    /// Fill `table[dst..]` with values from `items`
+    ///
+    /// Returns a trap error on out-of-bounds accesses.
+    pub fn init_funcs(
+        &self,
+        dst: u32,
+        items: impl ExactSizeIterator<Item = *mut VMCallerCheckedAnyfunc>,
+    ) -> Result<(), Trap> {
+        let ty = TableElementType::Func;
+        assert!(ty == TableElementType::Func);
```

Comment:
This seems very tautological as written; is this supposed to be checking `self.element_type`?

Reply:
Oops, yes!